sphinx/builders/linkcheck.py

```diff
@@ -36,28 +38,25 @@ class HTMLParseError(Exception):
 from sphinx.util import encode_uri
 from sphinx.util.console import purple, red, darkgreen, darkgray, \
     darkred, turquoise
-from sphinx.util.pycompat import TextIOWrapper
 
-
-class RedirectHandler(HTTPRedirectHandler):
-    """A RedirectHandler that records the redirect code we got."""
-
-    def redirect_request(self, req, fp, code, msg, headers, newurl):
-        new_req = HTTPRedirectHandler.redirect_request(self, req, fp, code,
-                                                       msg, headers, newurl)
-        req.redirect_code = code
-        return new_req
-
-# create an opener that will simulate a browser user-agent
-opener = build_opener(RedirectHandler)
-opener.addheaders = [('User-agent', 'Mozilla/5.0 (X11; Linux x86_64; rv:25.0) '
-                      'Gecko/20100101 Firefox/25.0')]
-
-
-class HeadRequest(Request):
-    """Subclass of urllib2.Request that sends a HEAD request."""
-    def get_method(self):
-        return 'HEAD'
+try:
+    pkg_resources.require(['requests[security]'])
+except pkg_resources.DistributionNotFound:
+    import ssl
+    if not getattr(ssl, 'HAS_SNI', False):
+        # don't complain on each url processed about the SSL issue
+        requests.packages.urllib3.disable_warnings(
+            requests.packages.urllib3.exceptions.InsecurePlatformWarning)
+        warnings.warn(
+            'Some links may return broken results due to being unable to '
+            'check the Server Name Indicate (SNI) in the returned SSL cert '
+            'against the hostname in the url requested. Recommended to '
+            'install "requests[security]" as a dependency or upgrade to '
+            'a python version with SNI support.'
```

Comment:
Just specify the known versions of python 3 and 2.7.9+, and if someone is running an older one it depends on whether the distro patched the support in but ignore that for the message?